### ISSUE TITLE
Unfriendly user experience when auto suggestions are provided but completions are blocked

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#5237]
In cases where `complete -x` blocks non-native completions, the ghosted autosuggestions retrieved from path context (not history) are probably more hurtful than helpful, as they imply that there are options available for completions but pressing `tab` yields nothing.

e.g. 

```fish
touch foo foo1 foo2
complete -c cat -ax bar
cat fo
```

fish will ghost autocomplete `foo` but the only way to select that is to use accept the commandline suggestion (arrow right). There's no way to access fish's knowledge of `foo1` or `foo2`, and if one is accustomed to accepting such a suggestion with TAB, that will fail.

**Comment by faho** [comment]
This just appears like a bug in that completion script to me, that should be fixed by removing that "-x" (and potentially replacing it with just a "-r").

**Comment by faho** [comment]
> the only sane option is to block file-based ghosted autocompletions from showing when -f/-x are in effect.

Wait... that's already supposed to be happening?

Can you retry your original problem? I've just tried, and after changing `-ax` to `-xa` (since the former errors out) with history disabled (`set -g fish_history ''`), I don't get any suggestion for `cat`.

**Comment by mqudsi** [comment]
You're right, I can't reproduce it with the `cat` example I provided. I couldn't get it to reproduce and was going to go crazy since I explored it thoroughly when I first experienced it, then I realized you already committed a fix for `git rebase --exec` (thank you!).

Using the git completions script as of 3b3361685a43f85b82455eb98cf811a5d0d20e0e, try the following:

```fish
mkdir temp
cd temp
git init
touch bar
git rebase -i master --exec b # but don't press enter here
```

At this point, `bar` will be autocompleted at the commandline but `bar` is not an option shown by the pager upon `TAB`. In fact, since in this case there are no completions matching `b` in this git repo, when you hit `TAB` you get that horrible terminal bell.

I've been extremely frustrated by this (as in, not able to just shrug it off) each time it's happened to me, but it makes sense that it's not a normal case because it doesn't happen to me often at all.

**Comment by mqudsi** [comment]
An example that works under `master`: `git show HEAD:` will autocomplete a path from the local folder, but tab is unavailable.

**Comment by faho** [comment]
>git show HEAD: will autocomplete a path from the local folder, but tab is unavailable

":", "=" and "@" are a bit special in `complete`. E.g `=` is used for `--opt=val`, and "val" needs to be completed separately.

So it seems that we're not suppressing files here correctly, yes. That is a _very_ special case though.

**Comment by mqudsi** [comment]
> So it seems that we're not suppressing files here correctly, yes. 

I don't know if you meant in the git completions or in the autosuggestions. If the latter, we're on the same page.

**Comment by faho** [comment]
In the suggestions, yes.

I've not been able to reproduce it without the ":" part, though!

**Comment by mqudsi** [comment]
`git checkout -- ` exhibits the same.

**Comment by bennofs** [comment]
Another example: `scp x:` shows path-based local autosuggestions which aren't selectable using TAB (and not correct either, since the part after `:` is a remote path)
